cpp/tests/binaryop/assert-binops.h

```diff
@@ -0,0 +1,211 @@
+    for (size_t i = 0; i < out_data.size(); ++i) {
+        ASSERT_EQ(out_data[i], (TypeOut)(op(lhs_data[i], rhs_h))) << i;
+    }
+
+    if (lhs.nullable()) {
+        ASSERT_TRUE(out.nullable());
+        auto lhs_valid = lhs_h.second;
+        auto out_valid = out_h.second;
+
+        uint32_t rhs_valid = (rhs.is_valid() 
+                            ? std::numeric_limits<bitmask_type>::max() : 0);
+        ASSERT_EQ(out_valid.size(), lhs_valid.size());
+        for (decltype(out_valid.size()) i = 0; i < out_valid.size(); ++i) {
+            ASSERT_EQ(out_valid[i], (rhs_valid & lhs_valid[i]));
+        }
+    } 
+    else {
+        if (rhs.is_valid()) {
+            ASSERT_FALSE(out.nullable());
+        } 
+        else {
+            auto out_valid = out_h.second;
+            for (decltype(out_valid.size()) i = 0; i < out_valid.size(); ++i) {
+                ASSERT_EQ(out_valid[i], 0);
+            }
+        }
+    }
+}
+
+template <typename TypeOut, typename TypeLhs, typename TypeRhs, typename TypeOp>
+void ASSERT_BINOP(column_view const& out,
+                  column_view const& lhs,
+                  column_view const& rhs,
+                  TypeOp&& op) {
+    auto lhs_h = cudf::test::to_host<TypeLhs>(lhs);
+    auto lhs_data = lhs_h.first;
+    auto rhs_h = cudf::test::to_host<TypeRhs>(rhs);
+    auto rhs_data = rhs_h.first;
+    auto out_h = cudf::test::to_host<TypeOut>(out);
+    auto out_data = out_h.first;
+
+    ASSERT_EQ(out_data.size(), lhs_data.size());
+    ASSERT_EQ(out_data.size(), rhs_data.size());
+    for (size_t i = 0; i < out_data.size(); ++i) {
+        ASSERT_EQ(out_data[i], (TypeOut)(op(lhs_data[i], rhs_data[i])));
+    }
+
+    if (lhs.nullable() and rhs.nullable()) {
+        ASSERT_TRUE(out.nullable());
+        auto lhs_valid = lhs_h.second;
+        auto rhs_valid = rhs_h.second;
+        auto out_valid = out_h.second;
+
+        ASSERT_EQ(out_valid.size(), lhs_valid.size());
+        ASSERT_EQ(out_valid.size(), rhs_valid.size());
+        for (decltype(out_valid.size()) i = 0; i < out_valid.size(); ++i) {
+            ASSERT_EQ(out_valid[i], (lhs_valid[i] & rhs_valid[i]));
+        }
+    }
+    else if (not lhs.nullable() and rhs.nullable()) {
+        ASSERT_TRUE(out.nullable());
+        auto rhs_valid = rhs_h.second;
+        auto out_valid = out_h.second;
+
+        ASSERT_EQ(out_valid.size(), rhs_valid.size());
+        for (decltype(out_valid.size()) i = 0; i < out_valid.size(); ++i) {
+            ASSERT_EQ(out_valid[i], rhs_valid[i]);
+        }
+    }
+    else if (lhs.nullable() and not rhs.nullable()) {
+        ASSERT_TRUE(out.nullable());
+        auto lhs_valid = lhs_h.second;
+        auto out_valid = out_h.second;
+
+        ASSERT_EQ(out_valid.size(), lhs_valid.size());
+        for (decltype(out_valid.size()) i = 0; i < out_valid.size(); ++i) {
+            ASSERT_EQ(out_valid[i], lhs_valid[i]);
+        }
+    }
+    else {
+        ASSERT_FALSE(out.nullable());
+    }
+    
```

Comment:
Is this empty line intended?

Reply:
Addressed in https://github.com/rapidsai/cudf/pull/3507/commits/ca001cf62e4e41859dd7ad75e0c5ff8ea53eebe0.